IdentityKit

ilyasmert opened the pull request.

PeeEmKay commented:
There are loads of implementations of this, is it not possible to write once and include / expose the method in each canister?


public query func icrc10_supported_standards() : async [SupportedStandard] {
        return [
            {
                url = "https://github.com/dfinity/ICRC/blob/main/ICRCs/ICRC-10/ICRC-10.md";
                name = "ICRC-10";
            },
            {
                url = "https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_28_trusted_origins.md";
                name = "ICRC-28";
            }
        ];
    };

ilyasmert commented:
When we connect with a wallet provider, we pass target canister ids to Identity Kit. And it first makes a query call to the icrc10_supported_standards method on each of the target canisters to check if that particular canister has the required standard. So, each of the target canisters must have this method.

For the icrc28 method calls, we need to return trusted origins based on local and UAT-PROD environments dynamically. So I implemented the `getTrustedOrigins` method in the `env.mo` files to avoid replication. Since the icrc10 method returns are static, I implemented the method directly. But it's possible to do the same for icrc10.

> There are loads of implementations of this, is it not possible to write once and include / expose the method in each canister?
> 
> public query func icrc10_supported_standards() : async [SupportedStandard] { return [ { url = "https://github.com/dfinity/ICRC/blob/main/ICRCs/ICRC-10/ICRC-10.md"; name = "ICRC-10"; }, { url = "https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_28_trusted_origins.md"; name = "ICRC-28"; } ]; };
